pkg/controller/elasticsearch/certificates/transport/reconcile.go

```diff
@@ -120,7 +122,7 @@ func ReconcileTransportCertificatesSecrets(
 func ensureTransportCertificatesSecretExists(
 	c k8s.Client,
 	es esv1.Elasticsearch,
-) (corev1.Secret, error) {
+) (*corev1.Secret, error) {
```

Comment:
I'm not very fond of returning pointers when not required (it introduces doubt: should we check for `nil`?), but for the sake of "just fixing the regression" I re-introduced the code as it was before https://github.com/elastic/cloud-on-k8s/commit/174e09b26b4feeed88104a75c5f56d82e0807b7f.